### PR TITLE
handle reconnect better to gregord CORE-3142

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco this implements the ping loop to get word about the connection being down from the daemon to gregord.

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: No visual changes found as a result of these commits.

File: go/libkb/config.go [hunk 1]
```diff
@@ -491,6 +491,10 @@ func (f JSONConfigFile) GetGregorSaveInterval() (time.Duration, bool) {
 	return f.GetDurationAtPath("push.save_interval")
 }
 
+func (f JSONConfigFile) GetGregorPingInterval() (time.Duration, bool) {
+	return f.GetDurationAtPath("gregor.ping_interval")
```
Comment (maxtaco) [inline go/libkb/config.go]: change this to `push.ping_interval`

File: go/libkb/env.go [hunk 2]
```diff
@@ -517,6 +518,13 @@ func (e *Env) GetGregorDisabled() bool {
 	)
 }
 
+func (e *Env) GetGregorPingInterval() time.Duration {
+	return e.GetDuration(10*time.Second,
+		func() (time.Duration, bool) { return e.getEnvDuration("GREGOR_PING_INTERVAL") },
```
Comment (maxtaco) [inline go/libkb/env.go]: ditto for this

File: go/libkb/env.go [hunk 3]
```diff
@@ -517,6 +518,13 @@ func (e *Env) GetGregorDisabled() bool {
...
+		func() (time.Duration, bool) { return e.config.GetGregorPingInterval() },
+	)
```
Comment (maxtaco) [inline go/libkb/env.go]: also can you check the command line?  To keep everything consistent everything can be configured in those 3 places.  Should be lots of examples to copy from....

**Comment by keybase-ci-visdiff** [comment]
:mag_right: No visual changes found as a result of these commits.

**Comment by maxtaco** [comment]
👍

**Pull request merged**